File: App.Contracts/Responses/RoomResponse.cs
using App.Domain.Entities;
using App.Domain.Enums;

namespace App.Contracts.Responses;

public sealed record RoomResponse(
    Guid Id,
    string RoomName,
    int RoomSize,
    int StartBid,
    int MinBid,
    int MaxBid,
    string ImgUrl,
    RoomStatus Status,
    int PlayersInRoom,
    IEnumerable<PlayerResponse> Players);

File: Web.ClientApp/src/contracts/room-response.ts
import {IPlayerResponse} from "./player-response";

export interface IRoomResponse{
    id: string;
    roomName: string;
    roomSize: number;
    startBid: number;
    minBid: number;
    maxBid: number;
    avatarUrl: string;
    status: string | number;  // TODO: fix type, must be only string
    playersInRoom: number;
    bank: number;
    players: Array<IPlayerResponse>;
}

